### DISCUSSION TITLE
Expose pixel to index function

### DISCUSSION THREAD

**Discussion by nick-keller** [TanStack/virtual#503] [Ideas]
First of all, congratulations on this great lib, very easy to work with.

## What is the feature request?
I'd like the `Virtualizer` to expose a method that would allow me to pass a value in pixels, and get back a `VirtualItem`.

## The story
I am the maintainer of [react-datasheet-grid](https://react-datasheet-grid.netlify.app/) which uses this lib to virtualize a 2D grid. I need to convert user clicks to cell coordinates.
- I cannot use `onClick` or other event listener on cells since the user might drag the mouse outside of the grid
- Not trivial to copmute with dynamic item size

## Conclusion
I ended up re-writing the core logic of this library just for this, and even considering re-implementing virtualization myself... :/
I feel like this feature sould be trivial to implement since the logic is already there under the hood.

Let me know what you tink!

**Comment by piecyk** [comment]
Hi @nick-keller,

> First of all, congratulations on this great lib, very easy to work with.

thanks 💪 

> I'd like the Virtualizer to expose a method that would allow me to pass a value in pixels, and get back a VirtualItem.

I think that's a valid request, right now we expose `getOffsetForIndex, getOffsetForAlignment` adding `getVirtualItemForOffset` sound good, checkout the https://github.com/TanStack/virtual/pull/505

**Reply by nick-keller** [reply]
Damn, congrats on the delivery, impressive work ;p
